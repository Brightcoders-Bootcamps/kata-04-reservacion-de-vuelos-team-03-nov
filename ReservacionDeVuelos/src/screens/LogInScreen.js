import React, {useState} from 'react';
import {View, Text, StyleSheet, TouchableOpacity} from 'react-native';
import colors from '../utils/Colors';
import contentText from '../utils/Constants';
import FormButton from '../components/FormButton';
import FormInput from '../components/FormInput';

const LogInScreen = ({navigation}) => {
  const [email, setEmail] = useState();
  const [password, setPasword] = useState();
  return (
    <View style={styles.container}>
      <View style={styles.ContainerInput}>
        <Text style={styles.title}> {contentText.logInTitle} </Text>
        <Text style={styles.formInputtext}> {contentText.email} </Text>
        <FormInput
          labelValue={email}
          onChangeText={(userEmail) => setEmail(userEmail)}
          secureTextEntry={false}
        />
        <Text style={styles.formInputtext}> {contentText.password} </Text>
        <FormInput
          labelValue={password}
          onChangeText={(userPassword) => setPasword(userPassword)}
          secureTextEntry={true}
        />
      </View>
      <FormButton
        buttonTitle="Log In"
        backgroundColor={colors.gray}
        color={colors.white}
        onPress={() => alert('Log In CLicked!')}
      />
      <View style={styles.login}>
        <Text style={styles.textLogIn}>{contentText.noAccount}</Text>
        <TouchableOpacity onPress={() => navigation.navigate('SignIn')}>
          <Text style={styles.goToSignIn}> {contentText.createHere} </Text>
        </TouchableOpacity>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: colors.white,
  },
  title: {
    fontSize: 20,
    color: colors.blue,
    alignItems: 'center',
    fontWeight: 'bold',
    marginBottom: 10,
  },
  ContainerInput: {
    marginTop: 25,
    marginBottom: 20,
  },
  formInputtext: {
    color: colors.black,
    fontSize: 15,
  },

  goToSignIn: {
    color: colors.blue,
    fontSize: 16,
    textDecorationLine: 'underline',
  },
  login: {
    flexDirection: 'row',
    marginTop: 40,
  },
  textLogIn: {
    fontSize: 16,
    color: colors.darkGray,
  },
});

export default LogInScreen;
