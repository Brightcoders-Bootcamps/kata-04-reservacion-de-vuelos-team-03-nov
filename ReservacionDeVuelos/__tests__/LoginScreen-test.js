// __tests__/Intro-test.js
import React from 'react';
import renderer from 'react-test-renderer';
import LogInScreen from '../src/screens/LogInScreen';

test('renders correctly', () => {
  const tree = renderer.create(<LogInScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});