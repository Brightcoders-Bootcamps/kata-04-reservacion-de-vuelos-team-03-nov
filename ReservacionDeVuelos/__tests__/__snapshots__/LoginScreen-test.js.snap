// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
      "padding": 20,
    }
  }
>
  <View
    style={
      Object {
        "marginBottom": 20,
        "marginTop": 25,
      }
    }
  >
    <Text
      style={
        Object {
          "alignItems": "center",
          "color": "#046BF1",
          "fontSize": 20,
          "fontWeight": "bold",
          "marginBottom": 10,
          "marginLeft": -5,
        }
      }
    >
       
      Log In
       
    </Text>
    <Text
      style={
        Object {
          "color": "#000",
          "fontSize": 15,
        }
      }
    >
       
      Email *
       
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderColor": "#000",
          "borderRadius": 3,
          "borderWidth": 1,
          "height": 88.93333333333334,
          "marginBottom": 10,
          "marginTop": 5,
          "width": 332,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        rejectResponderTermination={true}
        secureTextEntry={false}
        style={
          Object {
            "alignItems": "center",
            "borderColor": "#000",
            "color": "#333",
            "flex": 1,
            "fontSize": 16,
            "justifyContent": "flex-start",
            "padding": 10,
          }
        }
        underlineColorAndroid="transparent"
      />
    </View>
    <Text
      style={
        Object {
          "color": "#000",
          "fontSize": 15,
        }
      }
    >
       
      Password *
       
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderColor": "#000",
          "borderRadius": 3,
          "borderWidth": 1,
          "height": 88.93333333333334,
          "marginBottom": 10,
          "marginTop": 5,
          "width": 332,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        rejectResponderTermination={true}
        secureTextEntry={true}
        style={
          Object {
            "alignItems": "center",
            "borderColor": "#000",
            "color": "#333",
            "flex": 1,
            "fontSize": 16,
            "justifyContent": "flex-start",
            "padding": 10,
          }
        }
        underlineColorAndroid="transparent"
      />
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#C2C4C0",
        "borderRadius": 10,
        "height": 88.93333333333334,
        "justifyContent": "center",
        "marginTop": 10,
        "opacity": 1,
        "padding": 10,
        "width": 300,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#fff",
            "fontSize": 16,
            "fontWeight": "bold",
          },
          Object {
            "color": "#fff",
          },
        ]
      }
    >
      Log In
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "marginTop": 40,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#757575",
          "fontSize": 16,
        }
      }
    >
      Don't have an account? 
    </Text>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#046BF1",
            "fontSize": 16,
            "textDecorationLine": "underline",
          }
        }
      >
         
        Create here
         
      </Text>
    </View>
  </View>
</View>
`;
